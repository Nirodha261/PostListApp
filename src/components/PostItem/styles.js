/*
  Author: Nirodha Perera
  Date: 16/09/2019

  Defines styles for PostItem component
*/

import { StyleSheet } from 'react-native';

module.exports = StyleSheet.create({

  container: {
    flex: 1,
    paddingTop: 10,
    padding: 24,
    backgroundColor: '#ecf0f1'
  },
  box: {
    borderRadius: 8,
    padding: 24,
    backgroundColor: 'white',
    width: '100%'
  },
  shadow: {
    elevation: 5,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 10 },
    shadowOpacity: 0.3,
    shadowRadius: 4
  }

});