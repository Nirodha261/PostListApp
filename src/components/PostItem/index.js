
/*
 * Author: Nirodha Perera
 * Date: 16/09/2019
 *
 * Post item with details
 */
import React from 'React';
import {
  View,
  Text,
  TouchableOpacity
} from 'react-native';
import styles from './styles';
import CommonStyles from '../../styles/commonStyles';
import PropTypes from 'prop-types';

const PostItem = ({ userName, title, onPress }) => {
  return (
    <View style={[styles.container, CommonStyles.centerContent]}>
      <TouchableOpacity
        style={[styles.box, styles.shadow]}
        onPress={onPress}>
        <Text style={CommonStyles.boldText}>{userName}</Text>
        <Text style={CommonStyles.standardText}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
}

PostItem.propTypes = {
  userName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
};

export default PostItem;