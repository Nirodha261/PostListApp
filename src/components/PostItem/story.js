/*
  Author: Nirodha Perera
  Date: 17/09/2019

  Defines stories for PostItem component
*/

import React from "react";
import { storiesOf } from "@storybook/react-native";
import PostItem from './';

storiesOf("PostItem", module)
  .add("default view", () => (
    <PostItem
      userName={"test user name"}
      title={"test title"}
      onPress={() => console.log("action: PostItem on refresh")}
    />
  ));