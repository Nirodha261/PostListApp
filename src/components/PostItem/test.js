/*
  Author: Nirodha Perera
  Date: 17/09/2019

  Defines tests for PostItem component
*/
import React from 'react';
import PostItem from './'
import renderer from 'react-test-renderer';
const clickFn = jest.fn();

describe('PostItem Component', () => {

  it('PostItem component renders correctly', () => {
    const component = renderer.create(
      <PostItem
        userName={"test user name"}
        title={"test title"}
        onPress={clickFn}
      />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
