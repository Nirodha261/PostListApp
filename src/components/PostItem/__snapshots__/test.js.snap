// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem Component PostItem component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ecf0f1",
        "flex": 1,
        "padding": 24,
        "paddingTop": 10,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "elevation": 5,
        "opacity": 1,
        "padding": 24,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      test user name
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      test title
    </Text>
  </View>
</View>
`;
